examples/with-auth/components/session.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * A class to handle signing in and out and caching session data in sessionStore
+ *
+ * Note: We usewindow.XMLHttpRequest() here rather than fetch because fetch() uses
+ * Service Workers and they cannot share cookies with the browser session
+ * yet (!) so if we tried to get or pass the CSRF token it would mismatch.
```

Comment:
Note: I only used window.XMLHttpRequest to make the linter (xo) happy, I have since changed this in my own code to tell it XMLHttpRequest is a global so it doesn't trigger a warning.

Just to clarify, the session code still needs to use XMLHttpRequest (or something that uses it under the hood, like unfetch) rather than just fetch though, as browsers don't yet let support fetch reading existing http cookies (though not by design as part of some security design, it's just not supported yet but is slated as a future feature, at least by the Mozilla team).